src/storage/compaction_checker.cc

```diff
@@ -126,8 +126,10 @@ void CompactionChecker::PickCompactionFiles(const std::string &cf_name) {
     if (total_keys != 0 && delete_ratio > best_delete_ratio) {
       best_delete_ratio = delete_ratio;
       best_filename = iter.first;
-      best_start_key = std::move(start_key);
-      best_stop_key = std::move(stop_key);
+      best_start_key = start_key;
+      start_key.clear();
+      best_stop_key = stop_key;
+      stop_key.clear();
```

Comment:
@git-hulk by the way, would you mind check the logic here? Previously, it will move `rocksdb::Slice`, however, moving `rocksdb::Slice` will not clear it, so it's same as copy. L:108 has:

```c++
    if (start_key.empty() || stop_key.empty()) continue;
```

So it might break the logic. I've check the compaction, and personally I guess `clear` it is ok. Would you mind take a look?

Reply:
Yes, it's good to use `clear` here.